### PR TITLE
{vis}[foss/2023a] Visit v3.4.1, VTK v9.2.6, Qwt v6.3.0

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/349)

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 3 out of 4 (4 easyconfigs in total) cns1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/9809699e409421072213737388e84528 for a full test report.

Seems on generoso it can not clone git recursively

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+three-dimensional (2D and 3D) structured, adaptive and unstructured
+meshes. Owing to its customizeable plugin design, VisIt is capabable of
+visualizing data from over 120 different scientific data formats.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+sources = [{
+    'filename': 'v%(version)s.tar.gz',
+    'git_config': {
```
Comment (boegel) [inline easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb:27]: @pavelToman Can we avoid this and use the prepared https://github.com/visit-dav/visit/releases/download/v3.4.1/visit3.4.1.tar.gz as source tarball instead?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb (hunk 1 above)
Comment (pavelToman) [reply easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb:27]: After the change of sources to visit3.4.1.tar.gz there are some problem. It seems it needs download git recursively to build it by CMakeMake: https://visit-sphinx-github-user-manual.readthedocs.io/en/stable/building_visit/Building_Directly_With_CMake.html#

The `git lfs` is also required for reg. tests: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/test_suite.html#a-note-about-git-lfs

Comment (pavelToman) [reply easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb:27]: Should I add git-lft as a dependency?

Comment (pavelToman) [reply easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb:27]: I tried to install Visit with `visit3.4.1.tar.gz` but without reg. tests and it seems it is ok - I tested it by run gui on desktop cluster with `visit`

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/v/VTK/VTK-9.3.0-foss-2023a.eb [hunk 2]
```diff
@@ -57,7 +59,11 @@ configopts += "-DPython3_INCLUDE_DIR=$EBROOTPYTHON/include/python%(pyshortver)s
 configopts += "-DPython3_LIBRARY=$EBROOTPYTHON/lib/libpython%(pyshortver)s.so "
 # Other
 configopts += "-DVTK_USE_MPI=ON "
-configopts += "-DCMAKE_INSTALL_LIBDIR=lib"
+configopts += "-DCMAKE_INSTALL_LIBDIR=lib "
+configopts += "-DVTK_QT_VERSION=5 "
+configopts += "-DQt5_DIR=$EBROOTQT5 "
+configopts += "-DVTK_MODULE_ENABLE_VTK_GuiSupportQt=YES "
+configopts += "-DVTK_MODULE_ENABLE_VTK_ViewsQt=YES"
```
Comment (boegel) [inline easybuild/easyconfigs/v/VTK/VTK-9.3.0-foss-2023a.eb:66]: @pavelToman Can you open a separate PR for the changes being made here to the existing easyconfig for VKT 9.3.0, so it stands out a bit better?

File: easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    ('zlib', '1.2.13'),
+    ('Python', '3.11.3'),
+    ('Python-bundle-PyPI', '2023.06'),
+    ('SciPy-bundle', '2023.07'),
+    ('X11', '20230603'),
+    ('Mesa', '23.1.4'),
+    ('libglvnd', '1.6.0'),
+    ('Qt5', '5.15.10'),
+    ('Qwt', '6.3.0'),
+    ('VTK', '9.2.6'),
```
Comment (boegel) [inline easybuild/easyconfigs/v/Visit/Visit-3.4.1-foss-2023a.eb:40]: add a comment above to mention that Visit 3.4.1 requires VTK 9.2.x (9.3 doesn't work), ideally with a link that confirms that (maybe https://github.com/visit-dav/visit/issues/19547)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/v/VTK/VTK-9.3.0-foss-2023a.eb (hunk 2 above)
Comment (pavelToman) [reply easybuild/easyconfigs/v/VTK/VTK-9.3.0-foss-2023a.eb:66]: PR: https://github.com/easybuilders/easybuild-easyconfigs/pull/21221

Comment (boegel) [reply easybuild/easyconfigs/v/VTK/VTK-9.3.0-foss-2023a.eb:66]: see #21221

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3150.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/02e4844465aa8c05e453ace894965bf9 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @pavelToman!

**Pull request merged**